src/traccuracy/metrics/_ctc.py

```diff
@@ -253,3 +255,128 @@ def get_weighted_error_sum(
+        """For a given subgraph, computes the lengths of complete cell cycles by walking the
+        graph from one division to the next
+
+        Args:
+            subgraph (TrackingGraph): a TrackingGraph containing a single subgraph of
+                connected components
+            divs (list[Hashable]): a list of dividing nodes in the subgraph
+
+        Returns:
+            list[int]: a list of the lengths of complete cell cycles in the subgraph
+        """
+        lengths = []
+
+        # Collect list of direct daughters
+        daughters = []
+        for div in divs:
+            daughters.extend(list(subgraph.graph.successors(div)))
+
+        for node in daughters:
+            # Start length at 1 because we're already looking at the daughter
+            length = 1
+            succs = list(subgraph.graph.successors(node))
+            while len(succs) == 1:
+                length += 1
+                succs = list(subgraph.graph.successors(succs[0]))
+
+            # Must end in another division to be valid cycle
+            if len(succs) == 2:
+                lengths.append(length)
+
+        return lengths
+
+    def _get_cumsum(self, lengths: list[int], bins: np.ndarray) -> np.ndarray:
+        """Given a list of cell cycle lengths, computes cumulative sum from a normalized
+        histogram of the lengths
+
+        Args:
+            lengths (list[int]): a list of cell cycle lengths
+            bins (np.ndarray): bins for the histogram usually determined
+                by the max cell cycle length
+
+        Returns:
+            np.ndarray: an array the cumulative sum of the normalized histogram
+        """
+        # Compute track length histogram
+        hist, _ = np.histogram(lengths, bins=bins)
+
+        # Normalize
+        hist = hist / hist.sum()
+
+        # Compute cumsum
+        cumsum = np.cumsum(hist)
+
+        return cumsum
+
+    def _compute(self, data: Matched) -> dict[str, float]:
+        gt_lengths = self._get_lengths(data.gt_graph)
+        pred_lengths = self._get_lengths(data.pred_graph)
+
+        cca = self._get_cca(gt_lengths, pred_lengths)
+        return {"CCA": cca}
+
+    def _get_cca(self, gt_lengths: list[int], pred_lengths: list[int]) -> float:
+        """Compute CCA given two lists of cell cycle lengths
+
+        Args:
+            gt_lengths (list[int]): cell cycle lengths from the ground truth data
+            pred_lengths (list[int]): cell cycle lengths from the predicted data
+
+        Returns:
+            float: the cell cycle accuracy
+        """
+        # GT and pred must both contain complete cell cycles to compute this metric
+        if np.sum(gt_lengths) == 0 or np.sum(pred_lengths) == 0:
+            warnings.warn(
+                "GT and pred data do not both contain complete cell cycles. Returning CCA = 0",
+                stacklevel=2,
+            )
+            return 0
```

Comment:
Should we return `np.nan` here since we can't compute the metric?

Reply:
Yes you're right. That's what we're doing for other metrics so we should be consistent.